src/audiovideocontroller/DefaultAudioVideoController.ts

```diff
@@ -234,7 +243,8 @@ export default class DefaultAudioVideoController implements AudioVideoController
     this.meetingSessionContext.videoTileController = this._videoTileController;
     this.meetingSessionContext.videoDownlinkBandwidthPolicy = this.configuration.videoDownlinkBandwidthPolicy;
     this.meetingSessionContext.videoUplinkBandwidthPolicy = this.configuration.videoUplinkBandwidthPolicy;
-
+    this.meetingSessionContext.currentVideoSubscribeContext = new DefaultVideoSubscribeContext();
+    this.meetingSessionContext.previousVideoSubscribeContext = new DefaultVideoSubscribeContext();
```

Comment:
- I'm probably missing something. You are initializing `this.meetingSessionContext.previousVideoSubscribeContext` here and using it in `CreatePeerConnectionTask`. But you don't set `previousVideoSubscribeContext` or call any update APIs on it. What is `previousVideoSubscribeContext` for?

- Assume that you keep setting `previousVideoSubscribeContext` to `currentVideoSubscribeContext` per update completion. What happens if more than one new IndexFrame arrives SDK during the negotiation? i.e. `previousVideoSubscribeContext` stores just one history snapshot.
 
   > Make AudioVideoController update takes VideoSubscribeContext instead of sharing data in AudioVideoControllerState
   
   In the CR descipriton, a brief explanation about `currentVideoSubscribeContext`, `previousVideoSubscribeContext`, and the full lifcycle would be useful.

   e.g. I'm understanding in this way. Per update operation, SDK will be using an immutable set of the VideoSubscribeContext object until it's complete. What happen when a new IndexFrame is coming?

Reply:
> * I'm probably missing something. You are initializing `this.meetingSessionContext.previousVideoSubscribeContext` here and using it in `CreatePeerConnectionTask`. But you don't set `previousVideoSubscribeContext` or call any update APIs on it. What is `previousVideoSubscribeContext` for?

looked like I messed it up while rebasing

> * Assume that you keep setting `previousVideoSubscribeContext` to `currentVideoSubscribeContext` per update completion. What happens if more than one new IndexFrame arrives SDK during the negotiation? i.e. `previousVideoSubscribeContext` stores just one history snapshot.
>   > Make AudioVideoController update takes VideoSubscribeContext instead of sharing data in AudioVideoControllerState
>   

If track event is two negotiation behind, the issue would still exist. What's your suggestion ? I don't think there are timing guarantees from webrtc

>   
>   In the CR descipriton, a brief explanation about `currentVideoSubscribeContext`, `previousVideoSubscribeContext`, and the full lifcycle would be useful.
>   e.g. I'm understanding in this way. Per update operation, SDK will be using an immutable set of the VideoSubscribeContext object until it's complete. What happen when a new IndexFrame is coming?



![Untitled Diagram](https://user-images.githubusercontent.com/46945302/90084405-1ad6a300-dcca-11ea-9e2f-b8e5792360a8.jpg)